lib/mapping/CCampaignHandler.h

```diff
@@ -29,8 +29,12 @@ namespace CampaignVersion
 		RoE = 4,
 		AB = 5,
 		SoD = 6,
-		WoG = 6
+		WoG = 6,
+		VCMI = 1

```

Comment:
Perhaps use larger number than SoD (+some buffer for theoretical hota format)

Or even set vcmi format to '130' to match vcmi version (1.3.0)

Reply:
This number doesn't impact anything - it just should be unique in this particular build. It event shall not be preserved with releases.
Real numbers from json are compared with CampaignVersion::VCMI_MIN and CampaignVersion::VCMI_MAX

Reply:
In that case CampaignVersion::VCMI_MIN and CampaignVersion::VCMI_MAX may get such values.
'Format 130' is more meaningful than simple incrementation whenever it changes.
Don't see anything wrong with having (for example) format 130 followed by 140 or whatever vcmi version will be when this version gets bumped - they don't have to be continuous.

Not insisting on this though, so up to you.

Reply:
I don't get the idea: what should be maximum supported version for VCMI 1.3?

Reply:
VCMI 1.3 would use versions starting from 130 (and up to how many format changes we'll get during 1.3 development)

If we need to make format change in vcmi 1.4 it will immediately get bumped to '140', even if '130' ended up final format for 1.3

Reply:
so, how vcmi-1.3 will know whether it can support version 140 or not?

Reply:
Same as it is now. vcmi-1.3 will have VCMI_MAX = 130 (or 131/132/... depending on how many times format was changed during 1.3 development)

The only difference is that instead of keeping continious versioning for 1.4 format changes we'll bump format immediately to 140, skipping whatever left in 130-139 range

Reply:
it means, that vcmi-1.3 won't be able to open maps of version 140, which is not what we probably want.
I would thinking about not changing versions or changing it as rare as possible. Which means, that we may introduce couple new fields, but since 1.3 can open it - it still version 1.

Reply:
also, I don't think we want to have one more version we need to change frequently. At some point we may need to break backward compatibility, but hopefully it won't be soon.